### PR TITLE
Update: limit upload concurrency

### PR DESCRIPTION
**Pull request by wenboyu2**

### CODE REVIEW

File: src/components/ContentUploader/ContentUploader.js [hunk 1]
```diff
@@ -276,7 +278,7 @@ class ContentUploader extends Component<Props, State> {
     upload = () => {
         const { items } = this.state;
         items.forEach((uploadItem) => {
-            if (uploadItem.status !== STATUS_IN_PROGRESS) {
+            if (uploadItem.status === STATUS_PENDING) {
```
Comment (tonyjin) [inline src/components/ContentUploader/ContentUploader.js]: The previous logic would also allow uploads from `STATUS_ERROR` (which might be a valid case?) - `STATUS_COMPLETE` probably wouldn't be a valid starting point.

File: src/components/ContentUploader/ContentUploader.js [hunk 2]
```diff
@@ -67,6 +67,7 @@ type State = {
 
 const CHUNKED_UPLOAD_MIN_SIZE_BYTES = 52428800; // 50MB
 const FILE_LIMIT_DEFAULT = 100; // Upload at most 100 files at once by default
+const UPLOAD_CONCURRENCY = 6;
 
 class ContentUploader extends Component<Props, State> {
     id: string;
```
Comment (tonyjin) [inline src/components/ContentUploader/ContentUploader.js]: I think we're documenting all class properties here so `numItemsUploading:number = 0;` probably belongs here

File: src/components/ContentUploader/ContentUploader.js [hunk 3]
```diff
@@ -358,6 +367,7 @@ class ContentUploader extends Component<Props, State> {
         onUpload(item.boxFile);
 
         this.updateViewAndCollection(items);
+        this.upload();
```
Comment (tonyjin) [inline src/components/ContentUploader/ContentUploader.js]: Did you mean to add this here?

File: src/components/ContentUploader/ContentUploader.js [hunk 4]
```diff
@@ -419,6 +430,7 @@ class ContentUploader extends Component<Props, State> {
         });
 
         this.updateViewAndCollection(items);
+        this.upload();
```
Comment (tonyjin) [inline src/components/ContentUploader/ContentUploader.js]: Same question - did you mean to add this?

File: src/components/ContentUploader/ContentUploader.js (hunk 2 above)
Comment (wenboyu2) [reply src/components/ContentUploader/ContentUploader.js]: done

File: src/components/ContentUploader/ContentUploader.js (hunk 1 above)
Comment (wenboyu2) [reply src/components/ContentUploader/ContentUploader.js]: What'll happen when`STATUS_ERROR` is allowed is that the failed uploads will be retryed immediately, before any other pending items being uploaded. Which could potentially cause infinite retry.

File: src/components/ContentUploader/ContentUploader.js (hunk 3 above)
Comment (wenboyu2) [reply src/components/ContentUploader/ContentUploader.js]: Yes, this is for kicking off uploads for other pending items.
Upload is started when
- new items are added to the queue
- item upload succeeded
- item upload failed

File: src/components/ContentUploader/ContentUploader.js (hunk 4 above)
Comment (wenboyu2) [reply src/components/ContentUploader/ContentUploader.js]: yes, failed upload makes room for another item. Calling upload() to scan the items and find the next pending items to upload.

Comment (tonyjin) [reply src/components/ContentUploader/ContentUploader.js]: Got it

**Comment by wenboyu2** [comment]
@tonyjin Can you help merge? I no longer have write access 😄

**Pull request merged**